src/stapi_fastapi/routers/product_router.py

```diff
@@ -63,9 +63,28 @@ def __init__(
             tags=["Products"],
         )
 
+        self.add_api_route(
+            path="/order-parameters",
+            endpoint=self.get_product_order_parameters,
+            name=f"{self.root_router.name}:{self.product.id}:get-order-parameters",
+            methods=["GET"],
+            summary="Get order parameters for the product",
+        )
+
```

Comment:
We should probably throw a comment in here to explain what we are doing. Maybe:

```suggestion

        # We wrap `self.create_order` so we can explicitly parameterize `OrderRequest`
        # for this product. Unfortunately, we have to do by grabbing the specific
        # annotation and overriding it's type dynamically to satisfy pydantic. We
        # can't do this directly to `self.create_order` because doing it there changes
        # the annotation on every `ProductRouter` instance's `create_order`, not just
        # this one's.
```